src/blsprivatekey.hpp

```diff
@@ -42,6 +42,7 @@ class BLSPrivateKey {
     // Construct a private key from another private key. Allocates memory in
     // secure heap, and copies keydata.
     BLSPrivateKey(const BLSPrivateKey& k);
+    BLSPrivateKey(BLSPrivateKey&& k);
```

Comment:
What is the purpose of this?

Reply:
It's a move constructor, which allows the compiler to do proper return value optimization. It also speeds up all places were a move/swap could be used instead of a copy, e.g. what vectors internally do when enlarging the internal buffers.

The compiler usually creates a default move constructor, but I decided to make this explicit here as I want to be 100% sure that no unnecessary copies are done when moving/swapping.

When this constructor is called, it will swap the internal keydata with the given rvalue reference's keydata. It basically moves around ownership of keydata. You can also invoke this constructor if needed with `BLSPrivateKey newKey(std::move(oldKey))`, after which `newKey` will own the keydata and `oldKey` will be uninitialized (as it now owns the nullptr keydata). This is basically what the stl libraries do a lot and what the compiler also does to optimize stuff.